var/spack/repos/builtin/packages/sgpp/package.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Sgpp(SConsPackage):
+    """SGpp is a library and framework for sparse grids in different flavors.
+    SGpp supports both hierarchical spatially-adaptive sparse grids and the
+    dimensionally-adaptive sparse grid combination technique."""
+
+    homepage = "https://sgpp.sparsegrids.org"
+    url = "https://github.com/SGpp/SGpp/archive/v3.2.0.tar.gz"
+    maintainers = ['G-071', 'leiterrl', 'pfluegdk']
+
+    # Versions with Python 3 bindings:
+    version('master', git='https://github.com/SGpp/SGpp.git', branch='master')
+    version('3.3.0', sha256='ca4d5b79f315b425ce69b04940c141451a76848bf1bd7b96067217304c68e2d4')
+    version('3.2.0', sha256='dab83587fd447f92ed8546eacaac6b8cbe65b8db5e860218c0fa2e42f776962d')
+    # Versions with Python 2 bindings:
+    version('3.1.0', sha256='6b46bc5b3966e92567d6754130666bdffb7be1d1d2c1b427d7ce964b8eaab526')
+    version('3.0.0', sha256='4dd9049e664abd7db78c355fea5e192167812f443115d4bf686a51bb1e9bda9c')
+
+    # Patch that ensures libraries will actually
+    # be copied into prefix/lib upon installation
+    # (otherwise it would be prefix/sgpp/lib)
+    patch('directory.patch', when='@:3.2.0')
+    # Fix faulty setup.py in 3.2.0
+    patch('fix-setup-py.patch', when='@3.2.0')
+    # Backport opencl fix
+    patch('ocl.patch', when='@:3.2.0+opencl')
+
+    variant('simd',
+            default='sse3',
+            values=('sse3', 'sse42', 'avx', 'fma4', 'avx2', 'avx512'),
+            description='Specifies the SIMD extension to be used')
+    variant('python', default=True,
+            description='Provide Python bindings for SGpp')
+    variant('java', default=False,
+            description='Provide Java bindings for SGpp')
+    # export CLASSPATH=$(spack location --install-dir sgpp)/lib/jsgpp.jar
+    # After that one can compile and run simple java examples with sgpp
```

Comment:
Spack should automatically set `CLASSPATH` if you load the generated module or use `spack load` to load the package.

Reply:
This doesn't seem to work with the current version of this sgpp package. 

I played around with it a bit today and found out that I can get it to work if I use ```extends('openjdk', when='+java')``` instead of ```depends_on```. 
However, that alone is not sufficient. I also had to remove the line ```extends('python', when='+python')```, and only then the ```CLASSPATH``` was set automatically by ```spack load sgpp``` after installing it with ```spack install sgpp -python +java```.

Is this related to #987, or did I mess up somewhere?

Reply:
Yes, that's the same bug. You need to `extends('java', when='+java')`, but it breaks when a package extends multiple other packages.

Reply:
Is there any way around this bug?

Otherwise I would just remove the Java variant from the package until the bug is fixed. We have more people using the Python bindings, so I would favor those over the Java ones.

Reply:
I removed the java extension from the package in a6c9c67

Reply:
You have a few options:

1. Don't support Java bindings, no variant and no dep
2. Support it but without env vars set, just a dep but not an extends
3. Support it correctly, use dep but no extends, manually set env vars